### ISSUE TITLE
Regression with ditaa diagram in alpha.8

### ISSUE THREAD

**Issue by cexbrayat** [asciidoctor/asciidoctor-epub3#188]
I ran into this issue when giving a try to https://github.com/asciidoctor/docker-asciidoctor/pull/90 but it looks unrelated to asciidocotr v2, as I can reproduce in the docker image from master and bumping only `asciidoctor-epub3` from `alpha.7` to `alpha.8`.

To reproduce, you need a document with a ditaa diagram (this is an over-simplification of our document).

test.adoc
```adoc
= Test
:doctype: book
:imagesdir: images
:toc2:
:numbered:
:experimental:

include::chapter.adoc[]
```

chapter.adoc
```adoc
== Diagram
[ditaa, diagram, svg, align=center]
----
+-+
+-+
----
```

and then run:

     asciidoctor -b epub3 -r asciidoctor-epub3 -r asciidoctor-diagram test.adoc

(I run this in the docker image from master, with epub3 set to alpha.9).

The error looks like:

    asciidoctor: ERROR: chapter.adoc: image not found or not readable: /documents/images/d

Note that the error message displays an incorrect file name (`d` instead of `diagram`).

**Comment by cexbrayat** [comment]
I investigated a bit more and to be more accurate, it's the `alpha.8` version that introduced the regression (but we never caught it as the latest docker image was still using `alpha.7`).

@mojavelinux I scrolled through the commits included in `alpha.8` and https://github.com/asciidoctor/asciidoctor-epub3/commit/300689cb782d4355e5ac317d60e7a8b6488d7e31 is tweaking SVG images. Could that be related?

**Comment by cexbrayat** [comment]
I created a repo to make it easily reproducible https://github.com/cexbrayat/asciidoctor-epub-svg-issue-188

**Comment by mojavelinux** [comment]
Thanks. I'll have a look asap.

**Comment by mojavelinux** [comment]
Oh, I see what's happening. This stems from an incompatibility between Asciidoctor core and Asciidoctor Diagram. Core is assuming the image is being registered with an argument array instead of a single argument. Hence, when it tries to take the first item in the array, it instead gets the first char. I'll address this in the next patch release of core. Alas, it has nothing to do with Asciidoctor EPUB3 after all.

**Comment by mojavelinux** [comment]
See https://github.com/asciidoctor/asciidoctor/issues/3283.

**Comment by mojavelinux** [comment]
@cexbrayat A fix has been included in 2.0.9. Please retest when you get a chance.

**Comment by cexbrayat** [comment]
I confirm this is fixed by `2.0.9`, thanks @mojavelinux !
